internal/printer/print-to-tsx.go

```diff
@@ -60,86 +63,95 @@ func getTextType(n *astro.Node) TextType {
 func renderTsx(p *printer, n *Node) {
 	// Root of the document, print all children
 	if n.Type == DocumentNode {
+		hasChildren := false
 		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			if c.PrevSibling != nil && c.PrevSibling.Type == FrontmatterNode {
```

Comment:
It's unclear to me what this block of code is doing, could you add a comment?

Reply:
Sure thing!

Reply:
Added in [3432c8e](https://github.com/withastro/compiler/pull/518/commits/3432c8e4223ab5b86cdc605983ae01e0d1d6b81f)